core/space/services/services_fs.go

```diff
@@ -465,6 +465,61 @@ func (s *Space) AddItems(ctx context.Context, sourcePaths []string, targetPath s
 	return results, totalsRes, nil
 }
 
+func (s *Space) AddItemStream(ctx context.Context, fileName string, targetPath string, bucketName string) (domain.AddItemStreamResponse, error) {
+	err := s.waitForTextileInit(ctx)
+	if err != nil {
+		return domain.AddItemStreamResponse{}, err
+	}
+	path, err := os.Getwd()
+	fileLocalPath := path + "\\" + fileName
+	f, err := os.Open(path + "\\" + fileName)
+	if err != nil {
+		log.Error(fmt.Sprintf("error opening path %s", fileLocalPath), err)
+		return domain.AddItemStreamResponse{}, err
+	}
+	defer closeFile(f, fileLocalPath)
+
+	var targetPathBucket string
+	if targetPath == "" || targetPath == "/" {
+		targetPathBucket = fileName
+	} else {
+		targetPathBucket = targetPath + "/" + fileName
+	}
+	b, err := s.getBucketWithFallback(ctx, bucketName)
+	if err != nil {
+		return domain.AddItemStreamResponse{}, err
+	}
+	_, _, err2 := b.UploadFile(ctx, targetPathBucket, f)
```

Comment:
yeah doing that, it was giving an error previously that `no new variable on the left of :=` because of which i had put err2.
But now changed it to  `_, _, err = b.UploadFile(ctx, targetPathBucket, f) ` by replacing `:=` with just `=`